invenio_accounts/utils.py

```diff
@@ -216,3 +239,33 @@ def validate_username(username):
         # text explaining the validation rules.
         message = current_app.config["ACCOUNTS_USERNAME_RULES_TEXT"]
         raise ValueError(message)
+
+
+def validate_domain_form(form, field):
+    """Validator for use with WTForm."""
+    if not validate_domain(field.data):
+        raise ValidationError(_("The email domain is blocked."))
+
+
+def validate_domain(email):
+    """Validate the domain of email address."""
+    email = email.lower()
+    try:
+        prefix, domain = split_emailaddr(email)
+    except ValueError:
+        return False
+    with db.session.no_autoflush:
+        domain = current_datastore.find_domain(domain)
+        if domain is not None and domain.status == DomainStatus.blocked:
+            return False
+        return True
+
+
+def split_emailaddr(email):
+    """Split email address in prefix and domain."""
+    prefix, domain = email.split("@", 1)
```

Comment:
```suggestion
    prefix, domain = email.rsplit("@", 1)
```

To avoid bypassing with something like `"user@foo"@gmail.com`, which would return `foo"@gmail.com` as the domain (see [this SO answer](https://stackoverflow.com/a/12355882))